### PR TITLE
FSL : harmonization with SPM and cleaning up the project structure

### PR DESCRIPTION
**Pull request by thomasbtnfr**
**Important note: this PR is built on https://github.com/bids-standard/BEP028_BIDSprov/pull/85. https://github.com/bids-standard/BEP028_BIDSprov/pull/85 must be reviewed and merged first**

---

The purpose of this pull request is to **implement the same processes on the FSL parser as those in SPM**. This includes:
- Use of **UUID**
- Use of a **mapping for labels** via a different json file than SPM
- SPM and FSL now share the **same default graph** and the same **CONTEXT_URL**. In addition, **all the functions shared by the 2 parsers have been gathered in a `utils` file**. **Unit tests** have been written for these functions.
- The **overall structure of the project was reviewed**. It is now structured as follows:
  - bids_prov
     - fsl
        - fsl_config.py (specific configuration for FSL which is not in the utils file
        - fsl_labels.json
        - fsl_parser.py
     - spm
        - spm_config.py
        - spm_config.yml
        - spm_labels.json
        - spm_parser.py
     - tests
        - fsl and spm unit tests + graph inclusion
     - utils.py
     - visualize.py
  - results
  - .github
  - and some configuration files...

In addition, this pull request reworks the reading of FSL log files. Indeed, in accordance with this [issue](https://github.com/Inria-Empenn/BEP028_BIDSprov/issues/14), some lines were not read such as those containing numbers that stopped the reading for a particular step. Furthermore, lines starting with `did not find` were read and according to our discussion of 17/01, they should not be read. This has been corrected.

### CODE REVIEW

**Comment by thomasbtnfr** [comment]
Conflicts have been resolved. The PR is ready to be reviewed.

File: bids_prov/fsl/fsl_config.py [hunk 1]
```diff
@@ -15,6 +15,31 @@
 )
 
 
+def get_or_load(fn):
+    """
+    fn should return a json serializable object
```
Comment (cmaumet) [inline bids_prov/fsl/fsl_config.py:20]: Could you add a few words to state the goal of this function? And what is `fn`?

File: bids_prov/fsl/fsl_parser.py [hunk 2]
```diff
@@ -90,7 +80,7 @@ def readlines(filename: str) -> Mapping[str, List[str]]:
             # TODO : add </pre> as in
```
Comment (cmaumet) [inline bids_prov/fsl/fsl_parser.py:80]: Could we store this TODO in an issue to keep track of it?

**Review by cmaumet — APPROVED** [review]
This looks good! Thank you! I only have two minor comments and this can be merged.

**Comment by cmaumet** [comment]
(Note: I am merging this now so that I can more easily review the FSL examples --- @thomasbtnfr when you can would be great if you can address the two tiny comments above, thx!)

File: bids_prov/fsl/fsl_config.py (hunk 1 above)
Comment (thomasbtnfr) [reply bids_prov/fsl/fsl_config.py:20]: A [comment](https://github.com/Inria-Empenn/BEP028_BIDSprov/blob/boutiques/bids_prov/fsl/fsl_config.py#L20) has been added in the `boutiques` branch.

File: bids_prov/fsl/fsl_parser.py (hunk 2 above)
Comment (thomasbtnfr) [reply bids_prov/fsl/fsl_parser.py:80]: The following issue has been created : https://github.com/Inria-Empenn/BEP028_BIDSprov/issues/25

**Pull request merged**